### PR TITLE
improve release notes and best practices for upgrade experience

### PR DESCRIPTION
**Pull request by nslater**

### CODE REVIEW

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+.. NOTE::
+
+   A full restart upgrade cannot be done without downtime for your database.
+
+.. WARNING::
+
+   If you are using Docker, you must not run ``docker service update`` as this
+   performs a :ref:`rolling_upgrade`.
+
+   Instead you must remove the service and then recreate it.
```
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: can someone please verify that this is correct?

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    sh$ cp -R 1.0.6/config 1.1.2/config
+    sh$ cp -R 1.0.6/data 1.1.2/data
+
+If you're using custom plugins, you may also want to copy those over.
+
+Once this is done, you can switch from running ``1.0.6/bin/crate`` to running
+``1.1.2/bin/crate``.
+
+System Packages
```
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: can someone please tell me if the above manual section is correct, and whether the system packages handle do anything like this? (and if not, what do they do?

File: blackbox/docs/release_notes/1.2.x.txt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    sh$ mv logging.yml log4j2.properties
+
+System Properties
+-----------------
+
+If you have previously set or configured ``CRATE_JAVA_OPTS`` in your startup
+scripts or environment, you should remove this, as CrateDB can no longer be
+configured via this method.
+
+Configuration Changes
```
Comment (nslater) [inline blackbox/docs/release_notes/1.2.x.txt]: the ticket for this work had a subtask that said "rename changed and remove deprecated settings (also in custom init script if official packages are not used)" can someone pls verify that this section and the one previous properly address that

File: blackbox/docs/release_notes/1.2.x.txt [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+Heap Size
++++++++++
+
+If you have previously set or configured ``CRATE_MIN_MEM`` or ``CRATE_MAX_MEM``
+in your startup scripts or environment, you must remove both, and replace them
+with a single variable ``CRATE_HEAP_SIZE``. The ``CRATE_HEAP_SIZE`` variable
+sets both the minimum and maximum memory to allocate, and should be set to
+whatever your previous ``CRATE_MAX_MEM`` was set to.
+
+System Settings
```
Comment (nslater) [inline blackbox/docs/release_notes/1.2.x.txt]: I have collected all the changes from the Elasticsearch docs here. but I'm wondering if this covers Windows? or whether we need to?

**Comment by nslater** [comment]
@joemoe added you and @matriv as reviewers. if you think there's anyone else we need to add, please add them! :)

File: blackbox/docs/release_notes/1.2.x.txt [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+-----------
+
+You can no longer run CrateDB as the superuser on Unix-like systems. You should
+create a new ``crate`` user for running the CrateDB daemon.
+
+
+Logging
+-------
+
+Your logging config file must be renamed::
```
Comment (matriv) [inline blackbox/docs/release_notes/1.2.x.txt]: that's not correct, it's different format.....

File: blackbox/docs/release_notes/1.2.x.txt [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+.. NOTE::
+
+   If you bind to a network reachable IP address, you must follow the
+   instructions in the next section.
+
+Elasticsearch Bootstrap Checks
+------------------------------
+
+If you are binding to a network reachable IP address, the Elasticsearch
+component used by CrateDB runs a number of `bootstrap checks`_ during startup.
```
Comment (matriv) [inline blackbox/docs/release_notes/1.2.x.txt]: remove _

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 2 above)
Comment (joemoe) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: please check this with an engineer. everyone should be able to answer this. just pick one.

Comment (joemoe) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: i'm not sure if we should document it like this or more on a meta level. something like: "Extract new edition of CrateDB and make sure config and data folders are the same."

File: blackbox/docs/release_notes/1.2.x.txt [hunk 7]
```diff
@@ -0,0 +1,134 @@
+.. _1.2.x_series:
+
+====================
+1.2.x Release Series
```
Comment (joemoe) [inline blackbox/docs/release_notes/1.2.x.txt]: as discussed. :)

File: blackbox/docs/release_notes/1.2.x.txt [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+Many configuration settings and files have been renamed or removed. You must
+review the release notes for :ref:`version_1.2.0` and update your setup as
+necessary.
+
+SQL Changes
+-----------
+
+Several breaking changes were made to CrateDB's SQL. This includes changes to
+time parsing, syntax changes, and new reserved keywords. You must review the
```
Comment (joemoe) [inline blackbox/docs/release_notes/1.2.x.txt]: It wasn't that much, can we explicitly say what it was?

File: blackbox/docs/release_notes/1.2.x.txt [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+:ref:`configuration` file::
+
+    network.host: 0.0.0.0
+
+.. NOTE::
+
+   If you bind to a network reachable IP address, you must follow the
+   instructions in the next section.
+
+Elasticsearch Bootstrap Checks
```
Comment (joemoe) [inline blackbox/docs/release_notes/1.2.x.txt]: Do we always call them "`Elasticsearch` Bootstrap Checks"?

File: blackbox/docs/release_notes/1.2.x.txt [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+in your startup scripts or environment, you must remove both, and replace them
+with a single variable ``CRATE_HEAP_SIZE``. The ``CRATE_HEAP_SIZE`` variable
+sets both the minimum and maximum memory to allocate, and should be set to
+whatever your previous ``CRATE_MAX_MEM`` was set to.
+
+System Settings
++++++++++++++++
+
+You must manually configure some systems settings limits to ensure the smooth
+and efficient operation of Elasticsearch. For all the details why, check out the
```
Comment (joemoe) [inline blackbox/docs/release_notes/1.2.x.txt]: is it really `of Elasticsearch`? Shouldn't it be `of CrateDB`.

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+.. _full_restart_upgrade:
+
+====================
+Full Restart Upgrade
+====================
+
+The preferred way to upgrade a running CrateDB cluster is via the
+:ref:`rolling_upgrade` method. But this is not always an option (for example,
+when you are upgrading from one minor version to the next).
```
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: nope. minor to minor requires restart. it's only patch to patch that you can do a rolling upgrade

File: blackbox/docs/release_notes/1.2.x.txt (hunk 6 above)
Comment (nslater) [inline blackbox/docs/release_notes/1.2.x.txt]: the `_` here is required to turn this into a link (we link to the ES docs)

File: blackbox/docs/release_notes/1.2.x.txt (hunk 5 above)
Comment (nslater) [inline blackbox/docs/release_notes/1.2.x.txt]: fixed

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+Once you have upgraded every node on the cluster, you can start every node in
+the cluster.
+
+Manually
+++++++++
+
+From within the appropriate directory, run::
+
+    sh$ su - crate
+    sh$ bin/crate -d -p crate.pid
```
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: fixed

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+
+Next, prepare the 1.1.2 install by moving the default ``config`` directory out
+of the way::
+
+    sh$ mv 1.1.2/config 1.1.2/config.orig
+
+Now, copy over the 1.0.6 ``config`` and ``data`` directories::
+
+    sh$ cp -R 1.0.6/config 1.1.2/config
+    sh$ cp -R 1.0.6/data 1.1.2/data
```
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: fixed

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+Upgrade Every Node
+------------------
+
+.. WARNING::
+
+   Make sure that every CrateDB daemon has been stopped before you continue. If
+   you are following this process and you attempt to upgrade your cluster while
+   an nodes are still running, you may experience data corruption or data loss.
+
+   If you want to keep your cluster running while you upgrade, you must follow
```
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: fixed (removed)

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+    sh$ sudo systemctl stop crate.service
+
+Upgrade Every Node
+------------------
+
+.. WARNING::
+
+   Make sure that every CrateDB daemon has been stopped before you continue. If
+   you are following this process and you attempt to upgrade your cluster while
+   an nodes are still running, you may experience data corruption or data loss.
```
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: fixed

File: blackbox/docs/release_notes/1.2.x.txt (hunk 7 above)
Comment (nslater) [inline blackbox/docs/release_notes/1.2.x.txt]: fixed

File: blackbox/docs/release_notes/1.2.x.txt (hunk 8 above)
Comment (nslater) [inline blackbox/docs/release_notes/1.2.x.txt]: fixed

File: blackbox/docs/release_notes/1.2.x.txt (hunk 9 above)
Comment (nslater) [inline blackbox/docs/release_notes/1.2.x.txt]: as far as I know, these are checks *run by* Elasticsearch, not by CrateDB. but I do not think we actually document them in our docs anywhere

perhaps we should?

File: blackbox/docs/release_notes/1.2.x.txt (hunk 10 above)
Comment (nslater) [inline blackbox/docs/release_notes/1.2.x.txt]: in this case, yes it is specifically Elasticsearch, and that is where these changes are coming from. as to whether we want to be explicit about that I think depends on whether we're only ever going to be documenting them in this file (in which case I reckon let's keep the specificity). but if we plan on including a new page in the docs that explains "bootstrap checks" then yah sure we should probably start to talk about these things as something that CrateDB itself does

thoughts?

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 2 above)
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: @joemoe can you clarify which bit you want me to check with an engineer?

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 13 above)
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: I've added a new section below this. pls check

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 11 above)
Comment (matriv) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: yep ok, I misunderstood

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+the cluster.
+
+Manually
+++++++++
+
+From within the appropriate directory, run::
+
+    sh$ su - crate -c 'bin/crate -d -p crate.pid'
+
+This switches to the ``crate`` user, and then runs the ``bin/crate`` command.
```
Comment (matriv) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: This runs the ``bin/crate`` command as ``crate`` user.

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 17]
```diff
@@ -0,0 +1,175 @@
...
+
+If you're using a Debian or Ubuntu package, run::
+
+    sh$ sudo apt-get --only-upgrade install crate
+
+If you're running Red Hat, run::
+
+    sh$ yum update -y crate
+
+If you're using Arch Linux, you can download and install_ the latest `Arch Linux
```
Comment (matriv) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: remove _ from install

File: blackbox/docs/release_notes/1.1.1.txt [hunk 18]
```diff
@@ -11,17 +11,21 @@ Released on 2017/03/27.
    If you are upgrading a cluster, you must be running CrateDB 0.57.0 or higher
    before you upgrade to 1.1.1.
 
-   You cannot perform a :ref:`cluster_upgrade` to this version. Any upgrade to
-   this version will require a full cluster restart.
+   You cannot perform a :ref:`rolling_upgrade` to this version. Any upgrade to
+   this version will require a :ref:`full_restart_upgrade` and will will
```
Comment (matriv) [inline blackbox/docs/release_notes/1.1.1.txt]: 2 times `will`

File: blackbox/docs/release_notes/1.1.2.txt [hunk 19]
```diff
@@ -11,18 +11,23 @@ Released on 2017/04/10.
    If you are upgrading a cluster, you must be running CrateDB 0.57.0 or higher
    before you upgrade to 1.1.2.
 
-   If you want to perform a :ref:`cluster_upgrade`, your current CrateDB version
-   number must be :ref:`version_1.1.1` or higher. Any upgrade from a version
-   prior to this will require a full cluster restart.
+   If you want to perform a :ref:`rolling_upgrade`, your current CrateDB
+   version number must be :ref:`version_1.1.0`. If you want to upgrade from a
```
Comment (matriv) [inline blackbox/docs/release_notes/1.1.2.txt]: `1.1.1` version `1.1.0` was broken and not available in our distribution channels

File: blackbox/docs/release_notes/1.1.3.txt [hunk 20]
```diff
@@ -11,18 +11,23 @@ Released on 2017/05/09.
    If you are upgrading a cluster, you must be running CrateDB
    :ref:`version_0.57.0` or higher before you upgrade to 1.1.3.
 
-   If you want to perform a :ref:`cluster_upgrade`, your current CrateDB
-   version number must be at least :ref:`version_1.1.1`. Any upgrade from a
-   version prior to this will require a full cluster restart.
+   If you want to perform a :ref:`rolling_upgrade`, your current CrateDB
+   version number must be :ref:`version_1.1.0`. If you want to upgrade from a
```
Comment (matriv) [inline blackbox/docs/release_notes/1.1.3.txt]: also here `1.1.1`

File: blackbox/docs/release_notes/1.2.0.txt [hunk 21]
```diff
@@ -188,4 +193,128 @@ Fixes
...
+
+The default bind address has been changed from `0.0.0.0` to the loopback
+address (meaning it will only be accessible on ``localhost``).
+
+If you bind to a network reachable IP address (e.g. you're configuring your
+cluster in a production or staging environment and not just for development
+purposes on your local machine) you must add the following line to your
+:ref:`configuration` file::
+
+    network.host: 0.0.0.0
```
Comment (matriv) [inline blackbox/docs/release_notes/1.2.0.txt]: this is not 100% correct, I would somehow refer to _network.host label in configuration.txt page

File: blackbox/docs/release_notes/1.2.0.txt [hunk 22]
```diff
@@ -188,4 +193,128 @@ Fixes
...
+
+.. NOTE::
+
+   If you bind to a network reachable IP address, you must follow the
+   instructions in the next section.
+
+Elasticsearch Bootstrap Checks
```
Comment (matriv) [inline blackbox/docs/release_notes/1.2.0.txt]: I don't think we should reference Elasticsearch here and in this section.
@dobe what do you think?

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 23]
```diff
@@ -0,0 +1,175 @@
...
+
+.. _full_restart_upgrade:
+
+====================
+Full Restart Upgrade
+====================
+
+The preferred way to upgrade a running CrateDB cluster is via the
+:ref:`rolling_upgrade` method. But this is not always an option (for example,
+when you are upgrading from one minor version to the next).
```
Comment (dobe) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: would write this differently. so that only patch versions can be rollings. "minor" sounds small and therefore confusing for people not aware that this is actually the feature version. (or call it feature version)

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 24]
```diff
@@ -0,0 +1,175 @@
...
+
+When a :ref:`rolling_upgrade` is not possible, a full restart upgrade must be
+done if you want to update CrateDB.
+
+.. NOTE::
+
+   A full restart upgrade cannot be done without downtime for your database.
```
Comment (dobe) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: well, hm - obviously

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 1 above)
Comment (dobe) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: i would not start to mention specific distribution topics in this documentation

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 25]
```diff
@@ -0,0 +1,175 @@
...
+---------------
+
+The first step is to shut down every CrateDB node in your cluster. By this, we
+mean, shut down every instance of the CrateDB daemon. The host operating system
+can be left running.
+
+Manually
+++++++++
+
+To stop the CrateDB daemon manually, run::
```
Comment (dobe) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: hm, it seems that everything here is distribution dependent, why not just say you need to use the stop/start .... thingies as known in your actual setup?

File: blackbox/docs/best_practice/full_restart_upgrade.txt [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+.. WARNING::
+
+   Make sure that every CrateDB daemon has been stopped before you continue. If
+   you are following this process and you attempt to upgrade your cluster while
+   nodes are still running, you may experience data corruption or data loss.
+
+Once every node has been stopped, you can upgrade CrateDB.
+
+Manually
+++++++++
```
Comment (dobe) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: i would really consider seperating things like, "how do i stop start restart my cluster" which is totally setup dependent and "what do i need to do when i upgrade"

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 13 above)
Comment (dobe) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: well we should rather recommend not using the tarball directly, giving instructions here is really hard. i think anyone is using the tarball he should /will actually use a special data path and a special configuration, which means that the upgrade process is different. 

however @matriv is right, copying is not what usually is done.

Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: @dobe would you like me to cut down the specifics of the "manual" method to something a bit more high level and mention that we recommend using packages or docker instead? (this also addresses what @joemoe brought up)

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 17 above)
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: this makes the word "install" a link, which is correct

File: blackbox/docs/release_notes/1.2.0.txt (hunk 21 above)
Comment (nslater) [inline blackbox/docs/release_notes/1.2.0.txt]: @matriv I'm not exactly sure what you're proposing here. can you tell me how you would reword or edit this section?

File: blackbox/docs/release_notes/1.2.0.txt (hunk 22 above)
Comment (nslater) [inline blackbox/docs/release_notes/1.2.0.txt]: @joemoe mentioned this too

I guess it depends how the bootstrap checks appear to the end user and whether we plan to document in the main documentation as something to prepare for, and so people can learn about them without looking at this particular set release notes (I think we should)

in fact, the above ^^ suggests that maybe this specific section of the releases notes should be created as a standalone part of the documentation, and in fact, in these release notes, we say "1.2.0 introduced bootrstrap checks. please consult Bootstrap Checks for details and so you can make the necessary changes to your setup"

@dobe thoughts?

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 24 above)
Comment (nslater) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: it is obvious to us, sure. but I figure it might be a good idea to highlight this here so people who've never done one realise that the primary thing is they need to consider before attempting to do this. thoughts?

happy to remove if you don't think it's necessary

File: blackbox/docs/best_practice/full_restart_upgrade.txt (hunk 17 above)
Comment (matriv) [inline blackbox/docs/best_practice/full_restart_upgrade.txt]: ok. sorry!

File: blackbox/docs/release_notes/1.2.0.txt (hunk 21 above)
Comment (matriv) [inline blackbox/docs/release_notes/1.2.0.txt]: @nslater as discussed... Mention to have previous behaviour use 0.0.0.0 but also reference the conf section with the special values: _site_, _<network_if>_, etc...

**Comment by nslater** [comment]
@dobe I have stripped down the instructions in the "full restart upgrade" guide. because I'd like to get this imped fixed as soon as possible, I have created two internal tickets, that I can work on as stories in another sprint:

- a guide to start/stop on every supported platform
- a guide to updating the cratedb software on an individual operating system, taking care of the config and data directory and so on

**Comment by nslater** [comment]
@dobe @joemoe @matriv landed a bunch of changes that hopefully address all outstanding comments. have moved the bootstrap checks to a new guide under "best practices", so we're documenting this more generally (i.e. making this a regular part of the docs, not hidden away in a specific version number release notes page)

reviews welcome! please and thx!! :)

File: blackbox/docs/best_practice/bootstrap_checks.txt [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+================
+
+If you are binding to a network reachable IP address, CrateDB performs a number
+of bootstrap checks during startup. These checks examine your setup and will
+prevent startup if a problem is detected.
+
+This best practices guide is intended to help you configure your setup so that
+CrateDB passes the bootstrap checks and can perform optimally.
+
+If you are binding to the loopback address, the bootstrap checks will not be
```
Comment (matriv) [inline blackbox/docs/best_practice/bootstrap_checks.txt]: remove **be**

File: blackbox/docs/best_practice/bootstrap_checks.txt [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+run, but it is still a good idea to follow these instructions.
+
+System Settings
+===============
+
+Official Packages
+-----------------
+
+If you are using one of the official packages, all of the necessary operating
```
Comment (matriv) [inline blackbox/docs/best_practice/bootstrap_checks.txt]: We need to double check what happens for docker image and especially **vm.max_map_count**

File: blackbox/docs/best_practice/bootstrap_checks.txt [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+Linux users should also edit ``/etc/sysctl.conf`` and configure::
+
+    vm.max_map_count = 262144
+
+
+Garbage Collection
+==================
+
+CrateDB uses `the CMS garbage collector`_ and this should not be changed or
+else performance may be seriously degraded.
```
Comment (matriv) [inline blackbox/docs/best_practice/bootstrap_checks.txt]: Usage of G1 garbage collector can even result in corrupting date (Lucene open issue) so I would change the phrase and remove the or else part.

File: blackbox/docs/best_practice/rolling_upgrade.txt [hunk 30]
```diff
@@ -268,5 +269,5 @@ Step 4: Repeat
 
 Repeat step two, three, and four for all other nodes.
 
-.. _back up your data: https://crate.io/a/backing-up-and-restoring-crate/
 .. _accompanying notes: https://crate.io/a/upgrade-your-cluster-to-0-55/
+.. _back up your data: https://crate.io/a/backing-up-and-restoring-crate/
```
Comment (matriv) [inline blackbox/docs/best_practice/rolling_upgrade.txt]: Not referenced in this text, I think we should also include a note that backing up date before the rolling upgrade is highly advised...

File: blackbox/docs/best_practice/bootstrap_checks.txt (hunk 27 above)
Comment (nslater) [inline blackbox/docs/best_practice/bootstrap_checks.txt]: I think this is fine? "will not be run" sounds more natural to me than "will not run". the difference is that with "will not run" implies they happen outside of crate (technically true, because they happen with ES, but this isn't the framing we want to use) so "will not be run" implies crate is deciding not to run them

thoughts?

File: blackbox/docs/best_practice/bootstrap_checks.txt (hunk 28 above)
Comment (nslater) [inline blackbox/docs/best_practice/bootstrap_checks.txt]: theres's currently an internal ticket open for this

https://trello.com/c/eJtuuDby/586

File: blackbox/docs/best_practice/rolling_upgrade.txt (hunk 30 above)
Comment (nslater) [inline blackbox/docs/best_practice/rolling_upgrade.txt]: this is reference in the text. check out the top of the "Upgrade Process" section

File: blackbox/docs/best_practice/bootstrap_checks.txt (hunk 28 above)
Comment (matriv) [inline blackbox/docs/best_practice/bootstrap_checks.txt]: great thanks!

File: blackbox/docs/best_practice/bootstrap_checks.txt (hunk 27 above)
Comment (matriv) [inline blackbox/docs/best_practice/bootstrap_checks.txt]: hm yep, I think it's fine. To me "will not be run" sounds like someone else is running them (passive voice) but also I'm not a native speaker :-)

File: blackbox/docs/best_practice/rolling_upgrade.txt (hunk 30 above)
Comment (matriv) [inline blackbox/docs/best_practice/rolling_upgrade.txt]: the github diff confused me, sorry :-)

File: blackbox/docs/best_practice/bootstrap_checks.txt [hunk 31]
```diff
@@ -0,0 +1,67 @@
...
+If you are using one of the official packages, all of the necessary operating
+system configuration is handled for you.
+
+Tarball
+-------
+
+If you have installed CrateDB from a tarball, you must manually configure a
+number of operating system settings.
+
+Edit ``/etc/security/limits.conf`` and configure these lines::
```
Comment (dobe) [inline blackbox/docs/best_practice/bootstrap_checks.txt]: this is only is true for linux. i think you need to describe "what" needs to be done. and if you describe "how" then it should be OS specific. at least osx is different.

File: blackbox/docs/release_notes/1.0.3.txt [hunk 32]
```diff
@@ -11,24 +11,29 @@ Released on 2017/02/10.
    If you are upgrading a cluster, you must be running CrateDB 0.57.0 or higher
    before you upgrade to 1.0.3.
 
-   If you want to perform a :ref:`cluster_upgrade`, your current CrateDB version
-   number must be :ref:`version_1.0.0` or higher. Any upgrade from a version
-   prior to this will require a full cluster restart.
+   If you want to perform a :ref:`rolling_upgrade`, your current CrateDB
+   version number must be :ref:`version_1.0.0`. If you want to upgrade from a
```
Comment (dobe) [inline blackbox/docs/release_notes/1.0.3.txt]: "must be at least"

see also other occurrences of this pattern

File: blackbox/docs/release_notes/1.2.0.txt [hunk 33]
```diff
@@ -188,4 +193,79 @@ Fixes
...
       SELECT count(*) from t1, t2 order by 1
 
+
+Upgrade Guide
+=============
+
+Daemon User
+-----------
+
+You can no longer run CrateDB as the superuser on Unix-like systems. You should
```
Comment (dobe) [inline blackbox/docs/release_notes/1.2.0.txt]: calling the section "upgrade guide" and then starting to write "you can no longer" is somehow odd.  either you rename it to "Upgrade Notes" which i would prefer, it works - but if it is called guide it would mean i have to write the paras like: "ensure that your crate daemon is not running as the root user, since ..."

Comment (nslater) [inline blackbox/docs/release_notes/1.2.0.txt]: fixed

**Comment by nslater** [comment]
@dobe I have modified the bootstrap checks guide to give general instructions, and then a specific section for configuring linux

I would like to leave out any specific instructions for configuring os x or windows, because that will take additional research and experimentation and testing on my part, and I believe we need these release notes finished for the 2.0 release

so if you're okay with it, I will create a story card for this work to be done later
